lib/puppet/resource/catalog.rb

```diff
@@ -433,7 +464,9 @@ def to_data_hash
       'environment'  => environment.to_s,
       'resources' => @resources.collect { |v| @resource_table[v].to_data_hash },
       'edges'     => edges.   collect { |e| e.to_data_hash },
-      'classes'   => classes
+      'classes'   => classes,
+      'metadata'  => metadata_hash,
```

Comment:
We should not serialize if these are empty, to avoid sending extra fields to agents that don't support static catalogs.